### ISSUE TITLE
Percentile Calculation Bug?

### ISSUE THREAD

**Issue by pigubaoza** [montanaflynn/stats#80]
Given a slice `a := []float64{0, 300, 600}`
`stats.Percentile(a, 50)` should return 300. However it returns 150.

**Comment by pigubaoza** [comment]
i think it's fixed here
https://github.com/montanaflynn/stats/pull/72

**Comment by JojoFlower** [comment]
the fix has not been released yet @montanaflynn
given  a := []float64{1, 2, 3}
stats.Percentile(a, 50) should return 1.5 instead of 2 🙏

**Comment by montanaflynn** [comment]
I think this is just a case of two different percentile algorithm's. Do you have a reference to the percentile algo that produces 1.5? The output should be 2 according to numpy and wolframalpha:

```python
import numpy as np
a = np.array([1, 2, 3])
print(np.percentile(a, 50))
// 2
```

https://www.wolframalpha.com/input?i=50th+percentile+of+1%2C+2%2C+3

**Comment by pigubaoza** [comment]
yes i believe for {0, 300, 600}, it should be 300, and for {1, 2, 3}, it should be 2. 
@JojoFlower please correct us if it's wrong. 

@montanaflynn is the fix (along with all the other merged pull requests), ready to be released yet?
